plugin/keyring_vault/vault_keys_list.cc

```diff
@@ -0,0 +1,44 @@
+#include "vault_keys_list.h"
+
+namespace keyring
+{
+
+//The caller takes ownership of the key, thus it is
+//his resposibility to free the key	
+my_bool Vault_keys_list::get_next_key(IKey **key)
+{
+  *key= NULL;
+  if (size() == 0 || keys_iter == keys.end())
+    return TRUE;
+  *key = *(keys_iter++);
+  return FALSE;
+}
+
+my_bool Vault_keys_list::has_next_key()
+{
+  return size() != 0 && keys_iter != keys.end();
+}
+
+size_t Vault_keys_list::size()
+{
+  return keys.size();
+}
+
+Vault_keys_list::~Vault_keys_list()
+{
+  //remove not fetched keys
```

Comment:
In keyring framework the design is to use plural - for instance keys_container. If I change it to singular now it will break the naming convention from keyring framework.